test/test.js

```diff
@@ -0,0 +1,98 @@
+var chai = require('chai');
+var chaiHttp = require('chai-http');
+chai.use(chaiHttp);
+var expect = chai.expect;
+var mongoose = require('mongoose');
+
+var User = require('../models/user')('Test');
+var app = require('../app')('Test');
+var agent = chai.request.agent(app);
+
+describe('clear db before testing', () => {
+
+  before('Remove previous users', done => {
+    mongoose.connect('mongodb://localhost/passport_demo');
+    mongoose.connection.on('connected', () => {
+      mongoose.connection.collection('tests').drop( () => done() );
+    });
+  });
+
+  after('Close db connection', done => {
+    mongoose.connection.close( () => done() );
+  });
+
+  describe('protected routes', () => {
+    it('should redirect to home page', done => {
+      chai.request(app)
+      .get('/secret')
+      .redirects(0)
+      .end( (err, res) => {
+        expect(err).to.be.null;
+        expect(res).to.redirectTo('/');
+        done();
+      });
+    });
+  });
+
+  describe('registering a new user', () => {
+    it('should register a new user', function(done) {
+      this.timeout(5000);
+      agent
+      .post('/register')
+      .redirects(0)
+      .send({ username: 'george', password: 'abc' })
```

Comment:
Handy to create something like:

```js
const testUser = { username: 'george', password: 'abc' };
```

So you don't have to repeat hard-coded info across tests